contributors/devel/release.md

```diff
@@ -0,0 +1,246 @@
+
+The process for shepherding features, issues, and pull requests
+into a Kubernetes release spans multiple stakeholders:
+* the feature, issue, or pull request owner
+* SIG leadership
+* the release team
+Information on workflows and interactions are described below.
+
+As the owner of a feature, issue, or pull request (PR), it is your
+responsibility to ensure release milestone requirements are met.
+Automation and the release team will be in contact with you if
+updates are required, but inaction can result in your work being
+removed from the milestone.  Additional requirements exist when the
+target milestone is a prior release (see [cherry pick
+process](cherry-picks.md) for more information).
+
+In the past there was a requirement for a milestone targeted pull
+request to have an associated GitHub issue opened, but this is no
+longer the case.  Features are effectively GitHub issues (or KEPs)
+which lead to subsequent PRs. The general labeling process should
+be consistent across artifact types.
+
+---
+
+## Definitions
+
+- *issue owners*: Creator, assignees, and user who moved the issue into a release milestone.
+- *release team*: Each Kubernetes release has a team doing project
+  management tasks described
+  [here](https://git.k8s.io/sig-release/release-team/README.md).  The
+  contact info for the team associated with any given release can be
+  found [here](https://git.k8s.io/sig-release/releases/).
+- *Y days*: Refers to business days (using the location local to the release-manager M-F).
+
+## The Release Cycle
+
+![Image of one Kubernetes release cycle](release-cycle.png)
+
+Kubernetes releases currently happen four times per year.  The release
+process can be thought of as having three main phases:
+* Feature Definition
+* Implementation
+* Stabilization
+
+But in reality this is an open source and agile project, with feature
+planning and implementation happening at all times.  Given the
+project scale and globally distributed developer base, it is critical
+to project velocity to not rely on a trailing stabilization phase and
+rather have continuous integration testing which ensures the
+project is always stable so that individual commits can be
+flagged as having broken something.
+
+With ongoing feature definition through the year, some set of items
+will bubble up as targeting a given release.  The **feature freeze**
+starts ~4 weeks into release cycle.  By this point all intended
+feature work for the given release has been defined in suitable
+planning artifacts in conjunction with the Release Team's [features
+lead](https://github.com/kubernetes/sig-release/tree/master/release-team/role-handbooks/features).
+
+Implementation and bugfixing is ongoing across the cycle, but
+culminates in a code slush and code freeze period:
+* The **code slush** starts in week ~9 of the release cycle.  The master
+  branch only accepts PRs for a release milestone.  No additional feature
+  development is merged after this point.
```

Comment:
after this point = during code slush i.e. in week 9?

I've seen PRs with features (with the correct labels and milestone set) being merged just before the code freeze.

Reply:
I don't see this able to be more than a cultural request to project leadership to not merge late features, without moving to a high friction, high latency project management process and enforcing mechanisms.  So I'm trying to capture policy intent as it stands today, even if that isn't always matched by human actions or enforced by mechanism.

Reply:
> So I'm trying to capture policy intent as it stands today, even if that isn't always matched by human actions or enforced by mechanism.

:+1: